woocommerce/woocommerce-ios
# Product List: add a WIP banner

Fixes #1521 

## Changes

- Added a new `UIImage` icon constant from Gridicons `workInProgressBanner` + test case
- Added a new `UITableView` extension for adjusting the table header view height `updateHeaderHeight` similar to table footer view
- Created a static helper `UIView.createBorderView` to create a border view given height and color with default values
- In `TopBannerView`, added support for both actionable and non-actionable cases that hides certain views when the banner is not actionable
  - Added an initializer for `TopBannerViewModel` when the banner is not actionable
- In `ProductsViewController`, showed the WIP banner as the table header view with a frame-based container

## Testing

- Launch the app
- If the Products tab isn't visible, turn the Products feature on by going to Settings > Experimental Features > Products
- Go to Products tab --> should see the "Work in progress" banner at the top of the Product List

## Example screenshots

@kyleaparker I used the `gridicons-customize` icon from http://automattic.github.io/gridicons/, please let me know if there is another icon you prefer!

portrait | landscape
-- | --
![Simulator Screen Shot - iPhone 8 - 2019-11-26 at 10 17 15](https://user-images.githubusercontent.com/1945542/69591388-f486af80-1035-11ea-80ae-cda6d026906c.png) | ![Simulator Screen Shot - iPhone 8 - 2019-11-26 at 10 17 16](https://user-images.githubusercontent.com/1945542/69591407-05372580-1036-11ea-8ce4-eada031b3685.png)


Update release notes:
